03-scaling/knative/service-10.yaml

```diff
@@ -2,6 +2,7 @@ apiVersion: serving.knative.dev/v1alpha1
 kind: Service
 metadata:
   name: greeter
+  namespace: knativetutorial
   annotations:
     # Target 10 in-flight-requests per pod.
     autoscaling.knative.dev/target: "10"
```

Comment:
This annotation is in the wrong place but #74 fixes that. Just be aware a merge conflict may need to be resolved in this file given how close these two changes are together.

Reply:
Thanks for the heads up

Reply:
@lberk  @bbrowning - I usually avoid the need to add `namespace` to the resource files 

- all the scripts and commands in the tutorial has `--namespace` or `-n` to the kubectl command that ensures what NS they are deployed to 
- Also giving the users the flexibility to change the namespace if they wish to

Reply:
> This annotation is in the wrong place but #74 fixes that. Just be aware a merge conflict may need to be resolved in this file given how close these two changes are together.

this is already merged and  @lberk has rebased it

Reply:
@kameshsampath I can definitely appreciate that approach with allowing users to specify the namespace themselves, and I can also appreciate that this kind of change might land in the realm of bikeshedding.  However, when running through the tutorial I made multiple typo's (maybe that's just my own fault) leading to having to check where I'd applied the actual resource when things didn't work as advertised.  Even tab-completing didn't quite work that well because the namespace `knativetutorial` also competes with knative's own namespaces (serving, building, eventing, sources, etc).  If our goal for the tutorial is to make it as easy as possible for new users to get up to speed, I'd recommend specifying this in the yaml files, to make running through the tutorial as easy as possible.

Reply:
@lberk @kameshsampath I tend to agree with Lukas here, the less we make ppl type, the better. I for one copy-pasted the commands from the tutorial. It'd would have taken me twice the time if I had to type them.
What if we employ the default namespace feature of kubectl. Set it at the start, reset back to what it was in the end?
https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#setting-the-namespace-preference:
kubectl config set-context $(kubectl config current-context) --namespace=<insert-namespace-name-here>

Reply:
@lberk  @syedriko  - the `kubens` is your friend. I have quoted this tool in the pre-req and I use it heavily I am seeing it can also be installed via `krew`, so if you ppl are lazy typing namespaces then they an try setting it first ( I can add a tip to the doc) and then start typing other commands, IMHO using `--namespace` is always the k8s best practice for user clarity.

Reply:
@kameshsampath Thanks for the pointers to krew.dev and kubectx. However, I simply followed the tutorial and did not find a single mention of kubens' use and therefore did not use it. I even suggested to remove it from the prereqs in https://github.com/redhat-developer-demos/knative-tutorial/issues/80. Instead, I found many repetitive instructions for -n knativetutorial, which looked ripe for optimization.

Reply:
yeah thats my bad, as told will add `Tip` in the next release, though i dont want user to be forced to use that.

Reply:
@kameshsampath thanks as well for the notes about `kubens` but again, I'd reiterate that for the purposes of a tutorial, limiting the delta between their current setup and executing the contents as easily as possible is worth prioritising.  Are we teaching folks about knative? or krew, kubens, and knative and a variety of auxiliary tools?  If both @syedriko and I missed it, is it really that obvious to others?  Anyways, I'll rebase/fix the remaining sections as you suggest.

Reply:
i have added the tip in the latest doc changes about `kubens`, thats the hard part about Kubernetes - lots of thing need to be learned in parallel :( sorry about that